multisrc/src/main/java/eu/kanade/tachiyomi/multisrc/fansubscat/FansubsCat.kt

```diff
@@ -0,0 +1,417 @@
+            }
+        }
+    }
+
+    private fun parsePageListFromJson(response: Response): List<Page> {
+        val jsonObject = json.decodeFromString<JsonObject>(response.body.string())
+
+        return jsonObject["result"]!!.jsonArray.mapIndexed { i, it ->
+            Page(
+                i,
+                it.jsonObject["url"]!!.jsonPrimitive.content,
+                it.jsonObject["url"]!!.jsonPrimitive.content,
+            )
+        }
+    }
+
+    // Popular
+
+    override fun popularMangaRequest(page: Int): Request {
+        return GET("$apiBaseUrl/manga/popular/$page?hentai=$isHentaiSite", headers)
+    }
+
+    override fun popularMangaParse(response: Response): MangasPage = parseMangaFromJson(response)
+
+    // Latest
+
+    override fun latestUpdatesRequest(page: Int): Request {
+        return GET("$apiBaseUrl/manga/recent/$page?hentai=$isHentaiSite", headers)
+    }
+
+    override fun latestUpdatesParse(response: Response): MangasPage = parseMangaFromJson(response)
+
+    // Search
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        val filterList = if (filters.isEmpty()) getFilterList() else filters
+        val mangaTypeFilter = filterList.find { it is MangaTypeFilter } as MangaTypeFilter
+        val stateFilter = filterList.find { it is StateFilter } as StateFilter
+        val demographyFilter = filterList.find { it is DemographyFilter } as DemographyFilter
+        val genreFilter = filterList.find { it is GenreTagFilter } as GenreTagFilter
+        val themeFilter = filterList.find { it is ThemeTagFilter } as ThemeTagFilter
+        val builder = "$apiBaseUrl/manga/search/$page?hentai=$isHentaiSite".toHttpUrl().newBuilder()
+        mangaTypeFilter.addQueryParameter(builder)
+        stateFilter.addQueryParameter(builder)
+        demographyFilter.addQueryParameter(builder)
+        genreFilter.addQueryParameter(builder)
+        themeFilter.addQueryParameter(builder)
+        if (query.isNotBlank()) {
+            builder.addQueryParameter("query", query)
+        }
+        return GET(builder.toString(), headers)
+    }
+
+    override fun searchMangaParse(response: Response): MangasPage = parseMangaFromJson(response)
+
+    // Details
+
+    // Workaround to allow "Open in browser" to use the real URL
+    override fun fetchMangaDetails(manga: SManga): Observable<SManga> =
+        client.newCall(apiMangaDetailsRequest(manga)).asObservableSuccess()
+            .map { mangaDetailsParse(it).apply { initialized = true } }
+
+    // Return the real URL for "Open in browser"
+    override fun mangaDetailsRequest(manga: SManga): Request {
+        return GET("$baseUrl/${manga.url}", headers)
+    }
+
+    private fun apiMangaDetailsRequest(manga: SManga): Request {
+        return GET(
+            "$apiBaseUrl/manga/details/${manga.url.substringAfterLast('/')}?hentai=$isHentaiSite",
+            headers,
+        )
+    }
```

Comment:
```suggestion
    override fun mangaDetailsRequest(manga: SManga): Request {
        return GET(
            "$apiBaseUrl/manga/details/${manga.url.substringAfterLast('/')}?hentai=$isHentaiSite",
            headers,
        )
    }
    
    override fun getMangaUrl(manga: SManga): String {
        return "$baseUrl/${manga.url}"
    }
```
This kind of workaround is no longer needed as you can override `getMangaUrl`/`getChapterUrl` for correct webview links.